central/processindicator/datastore/datastore_impl.go

```diff
@@ -137,12 +142,28 @@ func (ds *datastoreImpl) removeMatchingIndicators(ctx context.Context, results [
 }
 
 func (ds *datastoreImpl) removeIndicators(ctx context.Context, ids []string) error {
+	log.Infof("Process Pruning")
 	if len(ids) == 0 {
 		return nil
 	}
 	if err := ds.storage.DeleteMany(ctx, ids); err != nil {
 		return err
 	}
+
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
```

Comment:
Keep an eye on this in scale and performance tests.  I tried some similar things with process indicators and network flows but in those scale environments I would get statement timeouts on these big deletes.  That is why DeleteMany batches.  It may be OK here, just something to keep in mind when testing.

Reply:
Good point, thanks. Generally speaking I believe it would be much better to rework pruning to do the cleanup by dropping old partitions, but it's quite a lot of work.

I've planned one round of perf evaluation for sql queries, so will check this out.